indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/StreamChunkParser.java

```diff
@@ -38,20 +45,38 @@
 {
   @Nullable
   private final InputRowParser<ByteBuffer> parser;
-  private final SettableByteEntityReader byteEntityReader;
+  private final Supplier<SettableByteEntityReader> lazyByteEntityReaderSupplier; // lazy initializer
 
-  StreamChunkParser(@Nullable InputRowParser<ByteBuffer> parser, SettableByteEntityReader byteEntityReader)
+  /**
+   * Either parser or inputFormat shouldn't be null.
+   */
+  StreamChunkParser(
+      @Nullable InputRowParser<ByteBuffer> parser,
+      @Nullable InputFormat inputFormat,
+      InputRowSchema inputRowSchema,
+      TransformSpec transformSpec,
+      File indexingTmpDir
+  )
   {
+    if (parser == null && inputFormat == null) {
+      throw new IAE("Either parser or inputFormat shouldn't be set");
+    }
     this.parser = parser;
-    this.byteEntityReader = byteEntityReader;
+    // Create a lazy initializer since it will fail to create a SettableByteEntityReader if inputFormat is null
+    this.lazyByteEntityReaderSupplier = Suppliers.memoize(() -> new SettableByteEntityReader(
```

Comment:
nit: Why a supplier instead of just making `byteEntityReader` `@Nullable`? it seems like it would not be used if `parser` is not null.

Reply:
Changed to nullable.